avocado/utils/process.py

```diff
@@ -291,8 +291,10 @@ class CmdResult(object):
     :param pid: ID of the process
     :type pid: int
     :param encoding: the encoding to use for the text version
-                     of stdout and stderr, with the default being
-                     Python's own (:func:`sys.getdefaultencoding`).
+                     of stdout and stderr. Usually
+                     :func:`sys.getdefaultencoding` is used but on python2
+                     we override "ascii" for "utf-8" as standard py2 does
+                     not report encoding properly and "utf-8" usually fits.
```

Comment:
I know it's hardcoded, that is why it fails to report the correct default encoding. Anyway the right solution should be found for `avocado.utils.astring.ENCODING` and then we can use it here (and we do have not about that). So for now I'd stick to this so Avocado is able to run again (as currently it does not work) and when we find the right solution we should use it everywhere (not just here).

Btw I looked at it and the best solution I found so far is `sys.stdout.encoding`, eventually followed by `LC_ALL` parsing if everything fails (on unix). Anyway this discussion is quite off topic of this PR, the intention of this PR is to make Avocado work with utf-8 outputs as it used to be able before the py3 fixes (due to fact that it did not care).

Reply:
I know why I keep confusing it. It's because when I was looking for this I found `sys.getfilesystemencoding()` and after the time I wrongly remembered it to be `sys.getdefaultencoding()`. With these in mind I think the `astring.ENCODING` should be based on `getfilesystemencoding` or `sys.stdout.encoding` (or combination) and the `sys.getdefaultencoding` should not be used at all.

Should I update this PR to have hotfix and then we can move to your PR to tweak the details, or do you insist on having the best shiny solution before merge?

Reply:
@ldoktor what you're proposing is a way for Avocado to choose an encoding based on various probes, since Python itself doesn't offer anything like that.  I'm adamant that, unless such a fail proof probe can be developed, the right thing to do is to have a default encoding, and, if there's demand for it, a configurable one.  Even the automatic (based on probes) can be an additional configurable option.

Let's recap: the current issue is that Avocado (on Python 2) won't work with non-ascii characters.  That's what we have to fix.  Automatically guessing/finding the right encoding is not the task at hand here.

Do you agree?

Reply:
@ldoktor there are issues on this PR, as proven by your confusion on how to elect an encoding described here: https://github.com/avocado-framework/avocado/pull/2538#discussion_r176073507.

On the other hand, the one issue which seems to be holding PR #2539 is that you **suggested** using `settings` instead of `defaults`.  So, I'm pretty sure which one of the PRs should be ready to be merged (and remember, most of the code on #2539 is yours anyway).